### DISCUSSION TITLE
How to work with multiple schemas in PGRST_DB_SCHEMAS via Swagger UI?

### DISCUSSION THREAD

**Discussion by Nick-Nal** [PostgREST/postgrest#2832] [Q&A] [answered]
Greetings

Just found out after some tests that Swagger UI connected to PostgREST always reads only the first schema from PGRST_DB_SCHEMAS list

Is there a way to make Swagger look into all schemas from PGRST_DB_SCHEMAS?

Or, alternatively, to add Content-Profile / Accept-Profile header as mandatory field to fill when forming request (defaulting to first / only schema from PGRST_DB_SCHEMAS)?

I'm thinking about adding these headers manually with PGRST_DB_PRE_REQUEST function choosing necessary schema for a user with certain JWT token, but that's not ideal because one user can have multiple available schemas to use, so it would be better to choose it from UI somehow

### Environment

* PostgreSQL version: Docker image postgres:15.3
* PostgREST version: Docker image postgrest/postgrest:v11.1.0
* Swagger UI version: Docker image swaggerapi/swagger-ui:v4.19.0

**Comment by laurenceisla** [comment, marked as answer]
Hi. Yes, this issue is currently open (https://github.com/PostgREST/postgrest/issues/1547). The only way to get the OpenAPI spec for another schema is to add the `Accept-Profile: <another_schema>` in the request, e.g.: 

```bash
curl "http://localhost:3000/" -H "Accept-Profile: <another-schema>"
```

But still, the Swagger UI will not add the Accept-Profile for any of the paths as you noticed. For now, we're not making big updates to OpenAPI due to what's mentioned here https://github.com/PostgREST/postgrest/issues/1698 (moving the OpenAPI implementation out of the core repo).

**Comment by Nick-Nal** [comment]
Thanks for clarification
I'll try to experiment with pre_request function for now or just get rid of OpenAPI (it's not vital for the app)

**Reply by laurenceisla** [reply]
There's also an option to override the whole OpenAPI spec using a function: https://postgrest.org/en/stable/references/api/openapi.html#overriding-full-openapi-response
